lib/galaxy/objectstore/cloud.py

```diff
@@ -0,0 +1,611 @@
+"""
+Object Store plugin for Cloud storage.
+"""
+
+import logging
+import multiprocessing
+import os
+import shutil
+import subprocess
+import threading
+import time
+
+from datetime import datetime
+
+from galaxy.exceptions import ObjectInvalid, ObjectNotFound
+from galaxy.util import (
+    directory_hash_id,
+    safe_relpath,
+    string_as_bool,
+    umask_fix_perms,
+)
+from galaxy.util.sleeper import Sleeper
+
+from ..objectstore import convert_bytes, ObjectStore
+from cloudbridge.cloud.factory import CloudProviderFactory, ProviderList
+
+try:
```

Comment:
`boto` was only used to handle some exceptions. However, I replaced all `boto`-specific exceptions with a generic exception. This is totally a temporary patch till exceptions are properly wrapped and thrown by `CloudBridge`.